### DISCUSSION TITLE
Rilis Baru v1.0-RC13: Incoming Webhook untuk Media

### DISCUSSION THREAD

**Discussion by rioastamal** [kirimwa-id/forum#111] [Pengumuman]
Halo semua,

Diakhir pekan bulan April 2022 ini saya dengan senang hati mengumumkan dirilisnya API KirimWA.id v1.0-RC13. Fitur-fitur utama yang dihadirkan pada versi kali ini adalah dukungan media untuk webhook incoming message dan juga filter untuk nomor pada webhook incoming webhook. Kesemuanya dapat disetting pada API `POST /v1/webhooks`.

## Incoming message pada media

Sekarang anda mempunyai pilihan untuk memproses media yang ada di _incoming message_. Dengan adanya pilihan ini maka anda bisa melakukan otomasi terhadap media tersebut. Contoh implementasinya adalah anda dapat membuat bot untuk melakukan manipulasi terhadap gambar yang masuk, misal melakukan resize atau filter grayscale berdasar caption yang dikirimkan. 

API KirimWA.id akan memberikan URL ke media tersebut dan anda dapat mendownload, memproses kemudian reupload atau yang lain. URL tersebut ada property `payload.metadata.media_url`. Maksimum total media yang diproses per bulan adalah **100MB**. 

Sekarang incoming message webhook mendukung semua tipe pesan yaitu text, image, document, video, contact, location.

Dokumentasi lengkap dapat merujuk di [https://developer.kirimwa.id/api/v1/#incoming-message-rules](https://developer.kirimwa.id/api/v1/#incoming-message-rules).

## Filter berdasarkan nomor pengirim

Dalam satu kasus anda mungkin tidak ingin menerima webhook dari setiap nomor. Anda dapat membatasi pengiriman webhook untuk nomor atau group tertentu saja. Misal anda hanya ingin menerima webhook jika ada pesan baru dari pengirim 621234567890 atau [Group ID](https://developer.kirimwa.id/api/v1/#groups) tertentu. Gunakan atribut allowed_senders pada saat memasukkan URL webhook melalui [POST /webhooks](https://developer.kirimwa.id/api/v1/#post-webhooks).

Dokumentasi lengkap dapat merujuk di [https://developer.kirimwa.id/api/v1/#incoming-message-rules](https://developer.kirimwa.id/api/v1/#incoming-message-rules).

## Cara Konfigurasi

Secara default fitur pemrosesan pada incoming message pada media tidak aktif. Begitu juga fitur filter berdasar nomor. Untuk mengaktifkannya gunakan API [POST /webhooks](https://developer.kirimwa.id/api/v1/#post-webhooks).

```sh
curl -X POST 'https://api.kirimwa.id/v1/webhooks' \
  -H 'Authorization: Bearer API_TOKEN' \
  -H 'Content-Type: application/json' \
  -d '
{
  "webhook_url": "https://YOUR_DOMAIN/my/webhook",
  "process_message_with_media": true,
  "allowed_senders": "621234567890,621234567890-1628639662"
}'
```

```json
{
  "id": "whu-YOUR@EMAIL",
  "status": "active",
  "data": "https://YOUR_DOMAIN/my/webhook",
  "incoming_message_rules": {
    "allowed_senders": [
      "621234567890",
      "621234567890-1628639662"
    ],
    "process_message_with_media": true
  },
  "created_at": "2021-06-20T17:17:06.333Z",
  "meta": {
    "location": "https://api.kirimwa.id/v1/webhooks"
  }
}
```

Saya harap dengan fitur ini akan mempermudah teman-teman developer dalam mengembangkan aplikasi yang mengandalkan API KirimWA.id.

Belum memiliki Token API KirimWA.id? Daftar gratis di https://developer.kirimwa.id/

Salam,
Rio Astamal

**Comment by kotak47** [comment]
terima kasih fitur barunya.. siap untuk dicoba
